### PR TITLE
[oneDNN] [tfg] [remapper] Add AsString + StringToHasBucketFast fusion in MLIR

### PR DESCRIPTION
**Pull request by kanvi-nervana**
This PR enables the following fusion and test in MLIR:

- AsString + StringToHashBucketFast -> _TensorToHashBucketFast

### CODE REVIEW

**Comment by gbaned** [comment]
Hi @rdzhabarov Can you please review this PR ? Thank you!

File: tensorflow/core/transforms/remapper/pass.cc [hunk 1]
```diff
@@ -179,11 +179,78 @@ static std::unique_ptr<OperationState> GetContractionBiasAddOpState(
...
+    // Not allowing control flow on op
+    if (helper_.HasControlOperandsOrResultUsers(op)) return failure();
+
+    if (op->getOperands().size() < 1) return failure();
+
+    TFOp stringtohash_wrapper(op);
+
+    // AsString op
+    Value as_string_value = op->getOperand(0);
+    Operation* as_string_op = as_string_value.getDefiningOp();
```
Comment (rdzhabarov) [inline tensorflow/core/transforms/remapper/pass.cc:200]: nit: just combine, op->getOperand(0).getDefiningOp();

File: tensorflow/core/transforms/remapper/tests/tensor_to_hashbucket.mlir [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+// CHECK-LABEL: tfg.func @tensor_to_hashbucket_test
+tfg.func @tensor_to_hashbucket_test() {
+  // CHECK: %[[PLACEHOLDER:.*]], {{.*}} name("input")
+  %Placeholder, %ctl = Placeholder device("/device:CPU:0") name("input") {dtype = i8, shape = #tf_type.shape<8x32x32x3>} : () -> (tensor<*xi8>)
+  // CHECK: %[[ASSTRING:.*]], {{.*}} name("to_string")
+  %AsString, %ctl_0 = AsString(%Placeholder) device("/device:CPU:0") name("to_string") {T = i8, fill = "", precision = -1 : i64, scientific = false, shortest = false, width = -1 : i64} : (tensor<*xi8>) -> (tensor<*x!tf_type.string>)
+  // CHECK: _TensorToHashBucketFast(%[[PLACEHOLDER:.*]]) {{.*}} name("to_bucket") {T = {{.*}}, num_buckets = {{.*}}}
+  %StringToHashBucketFast, %ctl_1 = StringToHashBucketFast(%AsString) device("/device:CPU:0") name("to_bucket") {num_buckets = 100 : i64} : (tensor<*x!tf_type.string>) -> (tensor<*xi64>)
+  return
+}
```
Comment (rdzhabarov) [inline tensorflow/core/transforms/remapper/tests/tensor_to_hashbucket.mlir:14]: fix new line

**Review by rdzhabarov — APPROVED** [review]
LGTM.

**Comment by gbaned** [comment]
Hi @rdzhabarov Can you please review this PR ? Thank you!

**Comment by gbaned** [comment]
Hi @rdzhabarov Can you please review this PR ? Thank you!

**Comment by gbaned** [comment]
Hi @rdzhabarov Can you please review this PR ? Thank you!

**Review by rdzhabarov — APPROVED** [review]

**Pull request merged**